### PR TITLE
Fix bugs inside demo

### PR DESCRIPTION
**Pull request by MakMuftic**
Fixes bugs:

- `TransactionTable` not using the injected extension and failing
- fix return value inside `getPolkadotAsset`
- fix `ts` errors
- Fix **snap** not installing and requesting permissions

### CODE REVIEW

File: packages/example/src/services/metamask.ts [hunk 1]
```diff
@@ -26,11 +26,15 @@ export const pluginOrigin = `wallet_plugin_${origin}`;
 
 export async function installPolkadotSnap(): Promise<boolean> {
     try {
-        await web3Enable('my cool dapp');
-        console.log("Snap installed!!");
+        await window.ethereum.send({
```
Comment (MakMuftic) [inline packages/example/src/services/metamask.ts:29]: @mpetrunic @irubido I am not sure if I am missing some information as I wasn't involved in this part of code as much, but the problem was that snap was never enabled and permissions requested. This works now but I am not sure if it is ok to use this code snip for enabling snap.

Comment (mpetrunic) [reply packages/example/src/services/metamask.ts:29]: nope, not ok, probably means enable function inside adapter isn't implemented properly

Comment (MakMuftic) [reply packages/example/src/services/metamask.ts:29]: @mpetrunic I see now, I will try few more things to solve this bug :)

File: packages/adapter/src/snap.ts [hunk 2]
```diff
@@ -64,8 +64,7 @@ export class MetamaskPolkadotSnap implements Injected {
   }
 
   public enableSnap = async (): Promise<Injected> => {
-
-    if(await isPolkadotSnapInstalled(this.pluginOrigin)) {
+    if(!(await isPolkadotSnapInstalled(this.pluginOrigin))) {
```
Comment (MakMuftic) [inline packages/adapter/src/snap.ts:67]: @mpetrunic @irubido :joy: this is why snap was never installing

**Pull request merged**